### ISSUE TITLE
Issue with 8.1.1 - JSPM Permission denied

### ISSUE THREAD

**Issue by nwhite89** [postcss/postcss-import#198]
I'm using `postcss-import` with webpack since the update to `8.1.1` and using the `pkg-resolve` module I am getting an error when running

```js
Unable to create jspm system folder
Error: EACCES, permission denied '/home/ubuntu/.jspm'
```

When I am running my project I don't have permissions to this folder and I believe it's unnecessary also I'm not entirely sure why `jspm` is even needed to be installed? It's not even used by `postcss-import`. For now to get this working I have had to downgrade to `8.1.0` any help on this would be greatly appreciated :). 

Thanks.

**Comment by MoOx** [comment]
Sorry for this @nwhite89. I am not even using jspm myself.

This is related to https://github.com/postcss/postcss-import/pull/194 that I accepted.
I guess it's because you ran into an undefined file, which end up in the catch handled by this PR.

@swernerx any idea how we can make the resolve optional? (only used in jpsm env)

**Comment by nwhite89** [comment]
@MoOx thanks for your quick reply what is the reasoning to use `pkg-resolve`? Looking at the code change I can't quite follow the necessity for using the package.

**Comment by MoOx** [comment]
It's to ensure JSPM support (but to be honest, I blindly trusted @swernerx, so we should wait for his input).

**Comment by swernerx** [comment]
Oops. That sounds "interesting". I never saw that issue here. 

Yep, pkg-resolve is for resolving jspm which is a pretty useful package manager for resolving the front-end part of typical webapps. 

I am not sure what happens there on your system that it throws such an error. JSPM seems to try to create the "~/.jspm" folder during runtime. The questions is why this raises an error at all. Shouldn't the app be able to create config files in the home directory?

JSPM uses the folder for user wide "global" caching of dependencies, etc. This is not implemented in "pkg-resolve" which is just querying "jspm" for its package registry.

**Comment by MoOx** [comment]
> EACCES, permission denied

Can be a "read" issue.

Is there a way to detect during runtime that we are in jspm env? Because if not, using pkg-resolve will only leads to error (eg: npm + postcss only). We should avoid that.

**Comment by swernerx** [comment]
It seems like JSPM is pretty intolerant with any unexpected situations regarding this folder. It throws with hard exceptions for any cases were:

- The home directory is not writable
- The directory exists but is not writeable
- There is a file which uses the same name instead of a directory.

I figure for me "jspm" should ignore all of these failed attempts and maybe present a warning instead of crashing.

**Comment by swernerx** [comment]
It does not seems like a "read" error but a "write" error. JSPM fails to create the folder ".jspm" in the home folder. I do not think that is exactly something wrong.

**Comment by swernerx** [comment]
@MoOx I am not sure what you mean by "jspm" environment... the error is all around a folder which should be created in the user's home directory. This is not project related as I understand.

**Comment by MoOx** [comment]
I mean: if I use npm only (not jspm), I won't expect jspm to throw an error.

**Comment by MoOx** [comment]
The pkg-resolve should not even be used if user is in a classic node/npm runtime.

**Comment by swernerx** [comment]
Sure, but the questions is how would you even figure out whether someone uses jspm...

**Comment by MoOx** [comment]
That's my question :)

**Comment by swernerx** [comment]
We could probably check whether the "package.json":

a) exists
b) has a "jspm" key

**Comment by swernerx** [comment]
I'll have a look whether I can improve the wrapping in "pkg-resolve" for that edge case.

**Comment by nwhite89** [comment]
Yeah that would be great, I kind of understand about JSPM failing hard if it actually is required. But for projects not using JSPM feels odd that is required when using postcss-import without JSPM.

**Comment by swernerx** [comment]
The problem come from just loading the "jspm" package. I think that such an hard error should probably never been thrown just from importing. That's why I opened an issue for JSPM-cli itself.

**Comment by swernerx** [comment]
I offered a pull request for jspm to fix the loading issue. I think for the moment we have to wait for them. So far I can see we can't fix this on our own as there is no conditional import in ES6.

**Comment by swernerx** [comment]
See also: https://github.com/jspm/jspm-cli/pull/1779

**Comment by swernerx** [comment]
I have a new pull request for postcss-import to fix the import issues. Just a workaround.

**Comment by swernerx** [comment]
https://github.com/postcss/postcss-import/pull/199/files

**Comment by nwhite89** [comment]
@swernerx thanks for the super quick turn around on this I saw your PR within JSPM has also be merged already and waiting on a release.

**Comment by MoOx** [comment]
PR are a bit weird, I am confused

**Comment by swernerx** [comment]
Check the new PR #205

**Comment by swernerx** [comment]
Sorry for the trouble with the previous attempts.

**Comment by MoOx** [comment]
Thanks! It's better than no attempts at all :)
